apps/qubit/modules/search/actions/autocompleteAction.class.php

```diff
@@ -76,7 +76,12 @@ public function execute($request)
 
         foreach ($items as $item) {
             $search = new \Elastica\Search($client);
-            $search->addIndex($index)->addType($index->getType($item['type']));
+            foreach ($indices as $type => $index) {
+                $elasticSearchTypeName = QubitSearch::getInstance()::ES_TYPE;
+
+                // This can be updated in ES 7.x when type params are optional
+                $search->addIndex($index)->addType($index->getType($elasticSearchTypeName));
+            }
```

Comment:
I found the issue, this is now adding all indices to all searches. We should get the wrapper before the loop, instead of getting all indices:

```php
$client = QubitSearch::getInstance()->client;
$indexWrapper = QubitSearch::getInstance()->index;
```

And only add the expected index to each search inside the loop:

```php
$search = new \Elastica\Search($client);
$search->addIndex($indexWrapper->getIndex($item['type']));
```

Reply:
And a couple of extra details after that:

- Changing this and the for loops in the initialize function makes `getIndices()` unnecessary in the index wrapper class.
- This should be noted as something that could break custom themes extending the autocomplete.